### ISSUE TITLE
[impl] computing in function

### ISSUE THREAD

**Issue by dannypsnl** [dannypsnl/elz#44]
At least:

- i32, i64, f32, f64: +, -, *, /
- return statement

**Comment by dannypsnl** [comment]
Need to parse return statement

**Comment by dannypsnl** [comment]
return statement at commit: 34f81794b53e009e1489bb360f51d69080cf9ce8

waiting for implementation of literal i32, f32, f64

For example:

```rust
fn main() {
  let i = 0 // i32
  let f = .1 // f32: 0.1
  let d = 1.2 as f64 // f64: 1.2
  let big_int = 5 as i64 // i64: 5
}
```

`as` expression require convertable & it's level is more tiny then `*`, `/`.

For example:

```rust
fn main() {
  let i = 3 as i64 - 6 as i64 // i64: -3
}
```

An do not allow any implict convert

For example:

```rust
fn main() {
  let i = 3 + 1.2 // panic, type error
  let c = 3 as f32 + 1.2 // this is correct one
}
```

**Comment by dannypsnl** [comment]
I want to drop `num` type

**Comment by dannypsnl** [comment]
Constant computing at commitL: bef6f47d77c4e22ef33aff48b5286d38fa3a8db7

Next:

- `as` expression for `i64`, `f64`
- function context, for local variable

**Comment by dannypsnl** [comment]
Fix a bug at: 54477579f7a96edf2387838f47c55df0a480e647

This bug's problem is we parsed `float` as `int`. It's a wrong implementation

**Comment by dannypsnl** [comment]
A little issue at here is we report reversed left & right expression at the Expression-Type internal report.

Fixed in commit: 702d6a542868898f14804688740e216d316fb1b2

**Comment by dannypsnl** [comment]
close this and have a new issue for computing content
